### PR TITLE
TEDEFO-2710 documentation about translations index and language-codes…

### PR DESCRIPTION
**Pull request by rouschr**
….xml files
As these files are new in SDK 1.10.x we need to document them and their potential impact on existing applications (especially language-codes.xml).

### CODE REVIEW

File: modules/translations/pages/index.adoc [hunk 1]
```diff
@@ -16,6 +16,72 @@ The texts provided are not final and are subject to change. In particular,
 languages other than English have often been obtained via automated translation, and have not been reviewed yet. We will continue to add and update procurement labels in every future release of the SDK.
 ====
 
+[#index]
+== Index file
```
Comment (bertrand-lorentz) [inline modules/translations/pages/index.adoc:20]: The term "index file" probably does not mean much for readers.
Please see how the equivalent file is described in the docs page for codelists or view-templates.

File: modules/translations/pages/index.adoc [hunk 2]
```diff
@@ -16,6 +16,72 @@ The texts provided are not final and are subject to change. In particular,
...
+
+`translations.json` is the translations folder index file, it contains two arrays documented below.
```
Comment (bertrand-lorentz) [inline modules/translations/pages/index.adoc:22]: We should not describe the "what", but rather explain the information provided. So something like:
"The file `translations.json` provides information on the translations files available in this folder, and the list of languages for which translations are provided."

File: modules/translations/pages/index.adoc [hunk 3]
```diff
@@ -26,6 +92,7 @@ field_en.xml
 ----
 
 The `<asset-type>` indicates the type of element the labels in the file are associated with. The possible asset types can be found in the xref:#asset-types[relevant section below]. The language identifier used is the standard 2 letter code in lowercase.
+There is a file named `language-codes.xml` (ending with .xml), this xref:#language-codes-xml[xml file] could be problematic for applications which do not expect the presence of this file, it is recommended to first go through the xref:#index[index file] found in the folder and get the desired xref:#files[filenames].
```
Comment (bertrand-lorentz) [inline modules/translations/pages/index.adoc:95]: This should be indicated as a [NOTE] (see the note at the top of the file), and could be a bit clearer by indicating that language-codes.xml does not contain translations and has a different format.

**Review by bertrand-lorentz — CHANGES_REQUESTED** [review]

File: modules/translations/pages/index.adoc (hunk 1 above)
Comment (rouschr) [reply modules/translations/pages/index.adoc]: There is one use of index file in understanding-the-sdk.adoc, but in general we used "Information on xyz" or just a sentence like: "The file named view-templates.json provides information on ...".
Both are fine, I think I will try to use "Information on xyz" as it better separates the files from the information about the files. Inside of the text we can try to call it the index file (but not in the title as you said).

File: modules/translations/pages/index.adoc (hunk 2 above)
Comment (rouschr) [reply modules/translations/pages/index.adoc]: I put something close to what you suggested.

File: modules/translations/pages/index.adoc (hunk 3 above)
Comment (rouschr) [reply modules/translations/pages/index.adoc]: It seems there are two ways to do a note, one that is more inline and another that is like a big block, I used the way it was done on top of the file, but I am not sure if I should use those ==== ... ==== around the text as those are not always used.
In the adoc docs they call this an "Admonition block", but it is not clear why we sometimes use it or not. Maybe just to better highlight where the note starts and ends?

**Review by bertrand-lorentz — APPROVED** [review]

**Pull request merged**